src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -908,26 +929,38 @@ public override ImmutableArray<CSharpAttributeData> GetAttributes()
 
             if (uncommon.lazyCustomAttributes.IsDefault)
             {
-                var loadedCustomAttributes = ContainingPEModule.GetCustomAttributesForToken(
-                    Handle,
-                    out _,
-                    // Filter out [Extension]
-                    MightContainExtensionMethods ? AttributeDescription.CaseSensitiveExtensionAttribute : default,
-                    out _,
-                    // Filter out [Obsolete], unless it was user defined
-                    (IsRefLikeType && ObsoleteAttributeData is null) ? AttributeDescription.ObsoleteAttribute : default,
-                    out _,
-                    // Filter out [IsReadOnly]
-                    IsReadOnly ? AttributeDescription.IsReadOnlyAttribute : default,
-                    out _,
-                    // Filter out [IsByRefLike]
-                    IsRefLikeType ? AttributeDescription.IsByRefLikeAttribute : default,
-                    out _,
-                    // Filter out [CompilerFeatureRequired]
-                    (IsRefLikeType && DeriveCompilerFeatureRequiredDiagnostic() is null) ? AttributeDescription.CompilerFeatureRequiredAttribute : default,
-                    out CustomAttributeHandle requiredHandle,
-                    // Filter out [RequiredMember]
-                    AttributeDescription.RequiredMemberAttribute);
+                ImmutableArray<CSharpAttributeData> loadedCustomAttributes;
+                CustomAttributeHandle requiredHandle;
+
+                if (IsExtension)
+                {
+                    // We do not recognize any attributes on extension blocks
```

Comment:
nit: consider leaving PROTOTYPE comments to cover the changes in `GetAttributes`, `CreateEvents`, `CreateFields` and `CreateNestedTypes` showing that attributes/events/fields/types in metadata are ignored.